clip/clip_score.py

```diff
@@ -28,17 +29,18 @@
 
 # Calculate the CLIP score for each image-caption pair
 clip_scores = []
-
+# Only compute CLIP Scores using generated images (To enable CLIP score computation for partial runs)
+id_list = [int(d.split(".")[0]) for d in os.listdir(args.image_folder)]
 for id, caption in zip(df['id'], df['caption']):
-    # Load the image
-    image_path = os.path.join(args.image_folder, f"{id}.png")
-    image = Image.open(image_path).convert("RGB")
-
-    # Calculate the CLIP score using the given class method
-    clip_score = clip_encoder.get_clip_score(caption, image)
-
-    # Store the CLIP score
-    clip_scores.append(clip_score.item())
+    # Check whether sample ID is generated
+    if id in id_list:
```

Comment:
Can we log a warning if some of the images are missing, or the total number of images that were evaluated ?

Other than this, the PR looks good.

Reply:
Thanks for adding this, I was about to suggest the change.